### DISCUSSION TITLE
`widthTracksTextView`, `heightTracksTextView` does not shrink to fit width/height

### DISCUSSION THREAD

**Discussion by sebjvidal** [krzyzanowskim/STTextView#35] [Questions]
STTextView's `widthTracksTextView` and `heightTracksTextView` works nicely when the size of textView needs to increase to fit its content. However, when deleting content, the size of the textView does not shrink to fit the content. See attached video.

https://github.com/krzyzanowskim/STTextView/assets/27621026/56ab6418-33d0-4f4b-b569-047925c07fb4

This can be reproduced with the following code on macOS 13.5.2 using Xcode 15.0 and STTextView 0.8.15:
```swift
import AppKit
import STTextView

class ViewController: NSViewController, STTextViewDelegate {

    override func viewDidLoad() {
        super.viewDidLoad()

        let textView = STTextView()
        textView.string = "Snippet"
        textView.backgroundColor = .systemTeal
        textView.font = .monospacedSystemFont(ofSize: 15, weight: .regular)
        
        view.addSubview(textView)
    }
}
```

**Comment by krzyzanowskim** [comment]
I've made major changes to that area. I'm happy to discuss if that's still an issue

**Comment by davbeck** [comment]
I was able to get a STTextView (inside a scroll view) working as a growing text field, but I ran into 1 main issue:

When you delete the last line, the size doesn't shrink back down. If you delete the last line again, it will shrink down, but still have the extra line height. This seems to be an issue with `textLayoutManager.usageBoundsForTextContainer.size`. I saw the exact same behavior with `NSTextView`. I'm not sure if this is a bug or a feature, but for this use case it gets in the way. Here's how I fixed it: https://github.com/davbeck/UntitledTerminalApp/blob/d40a2eddf50217a1ff2c1a23bbf99c7d0dc63298/Terminal/Views/Prompt/PromptTextView.swift#L23. I used `enumerateTextLayoutFragments` instead of `usageBoundsForTextContainer`.

**Comment by krzyzanowskim** [comment]
try 0.9.2, and in addition check values `isVerticallyResizable`, `isHorizontallyResizable` that configure shrinking behavior
